From 9270c7bdb35d4dce2c270592011498a12a1bf6e7 Mon Sep 17 00:00:00 2001
From: William Breathitt Gray <vilhelm.gray@gmail.com>
Date: Sat, 31 Aug 2019 21:23:23 +0900
Subject: [PATCH] Introduce the BUILD_NONFREE option

This allow users to disable building nonfree components (brightmaps.pk3,
game_support.pk3, and game_widescreen_gfx.pk3) if they so desire.
---
 CMakeLists.txt | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3763ec1f..d0c4720bd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -400,11 +400,14 @@ add_subdirectory( libraries/lzma )
 add_subdirectory( tools )
 add_subdirectory( libraries/gdtoa )
 add_subdirectory( wadsrc )
-add_subdirectory( wadsrc_bm )
 add_subdirectory( wadsrc_lights )
-add_subdirectory( wadsrc_extra )
-add_subdirectory( wadsrc_widescreen )
 add_subdirectory( src )
+option (BUILD_NONFREE "Build nonfree components" ON)
+if( BUILD_NONFREE )
+       add_subdirectory( wadsrc_bm )
+       add_subdirectory( wadsrc_extra )
+       add_subdirectory( wadsrc_widescreen )
+endif()
 
 if( NOT CMAKE_CROSSCOMPILING )
 	export(TARGETS ${CROSS_EXPORTS} FILE "${CMAKE_BINARY_DIR}/ImportExecutables.cmake" )
-- 
2.28.0

